src/stirling/bpf_tools/testdata/get_current_cgroup_id.c

```diff
@@ -0,0 +1,13 @@
+#include "src/stirling/bpf_tools/bcc_bpf/task_struct_utils.h"
+
+BPF_ARRAY(cgroup_id_output, uint64_t, 1);
+
+int probe_bpf_get_current_cgroup_id(struct pt_regs* ctx) {
+  uint64_t cgroup_id = UINT64_MAX;
+#if GET_CGROUP_ID_ENABLED
+  cgroup_id = bpf_get_current_cgroup_id();
+#endif
```

Comment:
While this `BCCWrapper` test verifies our compilation works, I think it would be better to have this `#if` check within library code. Otherwise we have to rewrite the same `#if` check in the socket tracer BPF code (meaning it's not benefitting from this test coverage).

My thinking is that we should add this "cgroup function" to `src/stirling/bpf_tools/bcc_bpf/utils.h`. Something along the lines of this:

```
// src/stirling/bpf_tools/bcc_bpf/utils.h

#ifndef GET_CGROUP_ID_ENABLED
uint64_t bpf_get_current_cgroup_id() {
  return UINT64_MAX;
}
#endif

```

Rather than only defining it for older kernels, I could also see a light wrapper making sense (like you did here).

We can then update `probe_bpf_get_current_cgroup_id` to include that `utils.h` header file and call that function. That will ensure that the version the socket tracer uses is exercised by this test..

Reply:
I see this makes a lot of sense. I'll make these changes.